cmd/syncer/leaderelectionutil.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"os"
+	"regexp"
+	"strings"
+)
+
+func defaultLeaderElectionIdentity() (string, error) {
+	return os.Hostname()
+}
+
+// sanitizeName sanitizes the provided string so it can be consumed by leader election library
+func sanitizeName(name string) string {
```

Comment:
Can we added context and debug logs to these new functions?

Reply:
Added a debug log for printing both identifier and namespace name
```
			log.Debugf("identity: %v, leaderElectionNamespace: %v",
				resourceLockConfig.Identity, *leaderElectionNamespace)
```